python/torch_mlir/eager/torch_dispatch.py

```diff
@@ -0,0 +1,353 @@
+
+    return s
+
+
+def build_script_function(
+    schema: torch._C.FunctionSchema,
+    args: List[torch._C.Argument],
+    kwargs: Dict[str, Any],
+) -> torch.jit.ScriptFunction:
+    """
+    Build a torch.jit.ScriptFunction that corresponds to the schema.
+    This is necessary so that we can pass a "hydrated" module to ModuleBuilder::importModule.
+    Hydrated meaning with inlined constants with correct values; a gotcha here is that if you don't inline the constants
+    then importModule assume default values (i think?) thereby (potentially) failing to support the op (not sure how this makes sense but native_batch_norm for example doesn't work without train=False)
+
+    Works by
+    1. creating an empty ts graph
+    2. creating and inserting a node corresponding to the op; nodes inserted are just names (hence following steps)
+    2. for each arg to the op:
+        a. if arg is a tensor: add input to the graph corresponding to arg
+           if arg is a constant (bool, int, list): inline a constant (at the top of the graph)
+        b. wire value from previous step to the op (ie SSA value corresponding to graph input is "passed" to op)
+        c. if arg is a tensor: fill in shape
+
+    The (poorly handled/poorly understood) gotcha here is args with type Optional[Tensors] with values None.
+    These pytorch types correspond to !torch.optional<!torch.vtensor>, which torch-mlir complains about having a torch.type_bound
+    (something something torch.type_bound can only be a func arg_attr for torch.vtensor). The hacky solution here works but
+    there's probably some straightforward way to handle this that I didn't stumble upon.
+
+    Parameters
+    ----------
+    schema: torch._C.FunctionSchema (i.e., c10::FunctionSchema)
+        Function in TorchScript IR to turn into MLIR.
+    args: ...
+
+    """
+    graph = torch._C.Graph()
+    node = graph.insertNode(graph.create(schema.name, len(schema.returns)))
+    for i, arg in enumerate(schema.arguments):
+        if arg.name in kwargs:
+            val = kwargs[arg.name]
+        else:
+            val = args[i]
+
+        if arg.type.isSubtypeOf(torch.TensorType.get()) or (
+            isinstance(arg.type, torch.OptionalType) and val is not None
+        ):
+            inp = graph.addInput()
+            if isinstance(arg.type, torch.OptionalType):
+                inp.setType(arg.type.getElementType())
+            else:
+                inp.setType(arg.type)
+        else:
+            inp = graph.insertConstant(val)
+            inp.node().moveBefore(node)
+
+        node.addInput(inp)
+
+    if node.hasMultipleOutputs():
+        for outp in node.outputs():
+            graph.registerOutput(outp)
+    else:
+        graph.registerOutput(node.output())
+
+    # this name will ultimately be the name of the func op in the mlir module and thus needs to be unique
+    # per op (since we keep the module around), hence smash together all of the things
```

Comment:
I guess this is necessarily valuable given that we're currently going with the "one module per op" approach...